Network-Office/src/main/java/dev/office/networkoffice/gathering/service/GatheringApplyService.java

```diff
@@ -0,0 +1,92 @@
+package dev.office.networkoffice.gathering.service;
+
+import java.util.List;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import dev.office.networkoffice.gathering.entity.Applicants;
+import dev.office.networkoffice.gathering.entity.Gathering;
+import dev.office.networkoffice.gathering.repository.ApplicantRepository;
+import dev.office.networkoffice.gathering.repository.GatheringRepository;
+import dev.office.networkoffice.user.entity.User;
+import dev.office.networkoffice.user.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+public class GatheringApplyService {
+
+	private final ApplicantRepository applicantRepository;
+	private final GatheringRepository gatheringRepository;
+	private final UserRepository userRepository;
+	private final GatheringAuthorityManagerService managerService;
+
+	/**
+	 * 모임 신청.
+	 * @param applicantId : 신청자 아이디
+	 * @param gatheringId : 신청하려는 모임 아이디
+	 */
+	@Transactional
+	public void applyGathering(Long applicantId, Long gatheringId){
+		User user = userRepository.findById(applicantId)
+			.orElseThrow(()-> new RuntimeException("유저없음"));//신청자 검증
+		Gathering gathering = gatheringRepository.findById(gatheringId)
+			.orElseThrow(()-> new RuntimeException("모임없음")); //모임검증
+		//기존 모임에 이미 user가 있는지 체크
+
+		//신청 엔티티 생성
+		Applicants applicants = Applicants.builder()
+			.gathering(gathering)
+			.user(user)
+			.build();
+		applicantRepository.save(applicants);
+	}
```

Comment:
```suggestion
	@Transactional
	public void applyGathering(Long applicantId, Long gatheringId){
             User user = getUserById(applicantId);
             Gathering gathering = getGatheringById(gatheringId);
		//기존 모임에 이미 user가 있는지 체크

		//신청 엔티티 생성
		Applicants applicants = Applicants.builder()
			.gathering(gathering)
			.user(user)
			.build();
		applicantRepository.save(applicants);
	}
	
	// ...
	
	private User getUserById(Long applicantId) {
		return userRepository.findById(applicantId)
				.orElseThrow(()-> new IllegalArgumentException("유저없음"));
	}

	private Gathering getGatheringById(Long gatheringId) {
		return gatheringRepository.findById(gatheringId)
				.orElseThrow(()-> new IllegalArgumentException("모임 없음"));
	}
```

- 사용자를 찾아오는 로직처럼 자주 사용되는 로직이라면 메서드 분리를 통해 가독성과 재사용성을 높이는 것은 어떨까요?
- 추가로 일단 분리했지만, `applicantId` 로 사용자를 찾는 것이 맞는 건가요?

Reply:
좋은 생각입니다!
> 추가로 일단 분리했지만, applicantId 로 사용자를 찾는 것이 맞는 건가요?
아닙니다, 유저Id로 찾아야하는데, 이곳저곳 한번에 바꾸다보니 섞여 들어간 것 같네요; 참고해서 수정해보겠습니다! 꼼꼼한 리뷰 덕분에 발견할 수 있었네요!☺️